spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -251,28 +250,24 @@ object DistriOptimizer {
               cached.modelGradients(i).zero()
             )
 
-            // copy multi-model gradient to the buffer
+            // Aggregate multi-model's gradient to the first model's gradient
             val parallelNum = if (taskSize == 0) extraTask else _subModelNumber
             Engine.default.invokeAndWait((0 until parallelNum).map(tid => () => {
               val offset = tid * taskSize + math.min(tid, extraTask)
               val length = taskSize + (if (tid < extraTask) 1 else 0)
-              var i = 0
+              var i = 1
               while (i < cached.modelGradients.length) {
-                if (i == 0) {
-                  cached.gradient.narrow(1, offset + 1, length)
-                    .copy(cached.modelGradients(i).narrow(1, offset + 1, length))
-                } else {
-                  cached.gradient.narrow(1, offset + 1, length)
+                  cached.modelGradients(0).narrow(1, offset + 1, length)
```

Comment:
If we directly aggregate other model's gradient to the first model's, there is a chance it's not correct if we turn on drop percentage as there is a possibility the first model's bp has not finished yet. I am not sure if we need consider this situation now.